src/mine_loop.rs

```diff
@@ -623,13 +624,15 @@ pub(crate) async fn mine(
     // their latest blocks. This should prevent the client from finding blocks that will later
     // be orphaned.
     const INITIAL_MINING_SLEEP_IN_SECONDS: u64 = 60;
-    tokio::time::sleep(Duration::from_secs(INITIAL_MINING_SLEEP_IN_SECONDS)).await;
-    let cli_args = global_state_lock.cli().clone();
 
     // Set PoW guessing to restart every N seconds, if it has been started. Only
     // the guesser task may set this to actually resolve, as this will otherwise
     // abort e.g. the composer.
     const GUESSING_RESTART_INTERVAL_IN_SECONDS: u64 = 20;
+
+    tokio::time::sleep(Duration::from_secs(INITIAL_MINING_SLEEP_IN_SECONDS)).await;
+    let cli_args = global_state_lock.cli().clone();
```

Comment:
Not sure why we need an owned version of CLI args here. Would a reference do?

Reply:
Maybe. It might cause borrowing issues later, when `global_state_lock.lock` is used while `cli_args` are still in scope. Note that (a) even though the diff suggests it, I did not introduce this line, and (b) `CliArgs` is pretty cheap to clone. Since this is not about any of the lints, I'll mark this conversation as resolved. Fee free to open a corresponding issue.